twelvelabs/resources/search.py

```diff
@@ -1,33 +1,64 @@
-from typing import List, Optional, Literal, Union, Dict, Any
+from typing import List, Optional, Literal, Union, Dict, Any, BinaryIO
 
 from ..resource import APIResource
 from .. import models
 from ..util import remove_none_values
 
+import json as jsonutil
+
 
 class Search(APIResource):
     def query(
         self,
         index_id: str,
-        query: Union[str, Dict[str, Any]],
         options: List[
-            Union[str, Literal["visual", "conversation", "text_in_video", "logo"]]
+            Literal["visual", "conversation", "text_in_video", "logo", "face"]
         ],
         *,
-        group_by: Optional[Union[str, Literal["video", "clip"]]] = None,
-        threshold: Optional[Union[str, Literal["high", "medium", "low"]]] = None,
-        operator: Optional[Union[str, Literal["or", "and"]]] = None,
-        conversation_option: Optional[
-            Union[str, Literal["semantic", "exact_match"]]
-        ] = None,
+        query: Union[str, Dict[str, Any]] = None,  # deprecated
```

Comment:
I suggest keeping `query` field for now but treat it as optional as it is a breaking change. 
My recommendation is that 
1. we document that query_text must be used instead, and also print warning message in SDK. 
2. Give notice to the users, 
3. and then along with our SDK major version bump like 90 days after, we remove this field.

Reply:
Yes, `query` is now optional field and will be removed after major release. it prints deprecated warning when using.